python/pyspark/rdd.py

```diff
@@ -135,12 +135,12 @@ def _load_from_socket(port, serializer):
         break
     if not sock:
         raise Exception("could not open socket")
-    try:
-        rf = sock.makefile("rb", 65536)
-        for item in serializer.load_stream(rf):
-            yield item
-    finally:
-        sock.close()
+    # The RDD materialization time is unpredicable, if we set a timeout for socket reading
+    # operation, it will very possibly fail. See SPARK-18281.
+    sock.settimeout(None)
+    # The socket will be automatically closed when garbage-collected.
+    rf = sock.makefile("rb", 65536)
+    return serializer.load_stream(sock.makefile("rb", 65536))
```

Comment:
Can you remove the `rf = sock.makefile("rb", 65536)` line since it is redundant?